packages/aws-rfdk/lib/deadline/lib/version.ts

```diff
@@ -72,6 +72,158 @@ abstract class VersionQueryBase extends Construct implements IVersion {
+    /**
+     * Regex: ^\d+(?:\.\d+){3}$
+     * Matches a sequence of '.' separated numbers with exactly 4 digits.
+     * - ^ asserts position at start of a line.
+     * - \d+ Matches one or more digits.
+     * - (?:\.\d+) Matches a dot and the following one or more digits.
+     * - {3} Matches previous pattern exactly 3 times.
+     * - $ asserts position at the end of a line
+     */
+    if (version.match(/^\d+(?:\.\d+){3}$/)) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Numeric components of version.
+   */
+  private readonly components: number[];
+
+  /**
+   * @inheritdoc
+   */
+  public get majorVersion(): number {
+    return this.components[0];
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public get minorVersion(): number {
+    return this.components[1];
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public get releaseVersion(): number {
+    return this.components[2];
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public get patchVersion(): number {
+    return this.components[3];
+  }
+
+  constructor(components: number[]) {
+    // validations
+    if(components.length != 4) {
+      throw new Error('Invalid version format. Version should contain exactly 4 components.');
+    }
+    components.forEach((component) => {
+      if (component < 0) {
+        throw new RangeError('Invalid version format. None of the version components can be negative.');
+      }
+    });
+
+    this.components = components;
+  }
+
+  /**
+   * This method compares two version strings
+   *
+   * @param version
+   *
+   * @returns true if this version is greater than the provided version;
+   * false if this version is less than or equal to the provided verison.
+   */
+  public isGreaterThan(version: Version): boolean {
+    if (this.components.length != version.components.length) {
+      throw new TypeError('Component count in both the versions should be same.');
+    }
+
+    return this.compare(version) > 0;
+  }
+
+  /**
+   * This method compares two version strings
+   *
+   * @param version
+   *
+   * @returns true if this version is less than the provided version;
+   * false if this version is greater than or equal to the provided verison.
+   */
+  public isLessThan(version: Version): boolean {
+    if (this.components.length != version.components.length) {
+      throw new TypeError('Component count in both the versions should be same.');
+    }
+
+    return this.compare(version) < 0;
+  }
+
+  /**
+   * The method returns the version components in dot separated string format.
+   */
+  public toString(): string {
+    return this.components.join('.');
+  }
+
+  /**
+   * This method compares 2 versions.
+   *
+   * @param version version to compare
+   *
+   * @returns negative value if this version is less than the provided version;
+   * 0 if both the versions are equal;
+   * positive value if this version is greater than the provided verison.
+   */
+  private compare(version: Version): number {
```

Comment:
We still don't have an easy way to compare for version equality.
Could you please add `isEqualTo` method similar to `isGreaterThan` and `isLessThan`?

Reply:
added.